### PR TITLE
[14.0] Add auth_api_key_server_env

### PR DESCRIPTION
**Pull request by simahawk**
Addresses https://github.com/OCA/server-auth/issues/273

ping @lmignon @dreispt

### CODE REVIEW

**Review by lmignon — APPROVED** [review]
GREAT! Thank you @simahawk. LGTM (code only)

File: auth_api_key/migrations/14.0.2.0.0/post-migrate.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+def migrate(cr, version):
+    if not version:
+        return
+
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    module = env["ir.module.module"].search(
+        [("name", "=", "auth_api_key_server_env"), ("state", "=", "uninstalled")]
+    )
+    if module:
+        module.write({"state": "to install"})
```
Comment (dreispt) [inline auth_api_key/migrations/14.0.2.0.0/post-migrate.py:22]: Nice trick!

File: auth_api_key_server_env/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+This can be very useful to avoid mixing your keys between your various
+environments when restoring databases. All you have to do is to add a new
+section to your configuration file according to the following convention:
+    """,
+    "version": "14.0.1.0.0",
+    "development_status": "Alpha",
+    "license": "LGPL-3",
+    "website": "https://github.com/OCA/server-auth",
+    "author": "Camptcamp,Odoo Community Association (OCA)",
```
Comment (dreispt) [inline auth_api_key_server_env/__manifest__.py]:
```suggestion
    "author": "Camptocamp,Odoo Community Association (OCA)",
```

File: auth_api_key_server_env/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    "depends": ["auth_api_key", "server_environment"],
```
Comment (dreispt) [inline auth_api_key_server_env/__manifest__.py:19]: Adding `auto_install : True` would be nice

**Review by dreispt — APPROVED** [review]

File: auth_api_key_server_env/__manifest__.py (hunk 3 above)
Comment (lmignon) [reply auth_api_key_server_env/__manifest__.py:19]: No auto_install.... IMO it's a choice from the dev/ana to store the api keys into server env even if others informations are managed in this way.

Comment (simahawk) [reply auth_api_key_server_env/__manifest__.py:19]: yep, that's the reason ;)

**Pull request merged**